src/container.rs

```diff
@@ -153,6 +154,70 @@ impl<T, const N: usize> ContainerExactly<T, N> for [T; N] {
     }
 }
 
+unsafe impl<T, C> ContainerExactly<T> for Box<C>
+where
+    C: ContainerExactly<T>,
+{
+    const LEN: usize = C::LEN;
+    type Uninit = Box<C::Uninit>;
+    fn uninit() -> Self::Uninit {
+        Box::new(C::uninit())
+    }
+    fn write(uninit: &mut Self::Uninit, i: usize, item: T) {
+        C::write(&mut *uninit, i, item)
+    }
+    unsafe fn drop_before(uninit: &mut Self::Uninit, i: usize) {
+        C::drop_before(&mut *uninit, i)
+    }
+    unsafe fn take(uninit: Self::Uninit) -> Self {
+        Box::from_raw(Box::into_raw(uninit) as *mut C)
+    }
+}
+
+unsafe impl<T, C> ContainerExactly<T> for Rc<C>
+where
+    C: ContainerExactly<T>,
+{
+    const LEN: usize = C::LEN;
+    type Uninit = Rc<UnsafeCell<C::Uninit>>;
+    fn uninit() -> Self::Uninit {
+        Rc::new(UnsafeCell::new(C::uninit()))
+    }
+    fn write(uninit: &mut Self::Uninit, i: usize, item: T) {
+        // SAFETY: We're the only owners of the uninit data at this point
+        C::write(unsafe { &mut *uninit.get() }, i, item)
+    }
+    unsafe fn drop_before(uninit: &mut Self::Uninit, i: usize) {
+        // SAFETY: We're the only owners of the uninit data at this point
+        C::drop_before(unsafe { &mut *uninit.get() }, i)
+    }
+    unsafe fn take(uninit: Self::Uninit) -> Self {
+        Rc::from_raw(Rc::into_raw(uninit) as *mut C)
+    }
+}
+
+unsafe impl<T, C> ContainerExactly<T> for Arc<C>
```

Comment:
I guess we should eventually add similar impls for `Container`, but no need in this PR